src/unix/os390.c

```diff
@@ -357,13 +356,11 @@ uint64_t uv_get_total_memory(void) {
 
 
 int uv_resident_set_memory(size_t* rss) {
-  char* psa;
   char* ascb;
   char* rax;
   size_t nframes;
 
-  psa = PSA_PTR;
-  ascb  = *(char* __ptr32 *)(psa + PSAAOLD);
+  ascb  = *(char* __ptr32 *)(PSAAOLD);
```

Comment:
The z/OS code is written as offsets into defined data structures (i.e. the base address of the structure+offset).
e.g. https://github.com/libuv/libuv/blob/0c2b3de331a106acf8e41c4a886b6c5eefa83245/src/unix/os390.c#L367
`ASCBRSME` is an offset into the Address Space Control Block (ASCB): 
https://www.ibm.com/support/knowledgecenter/en/SSLTBW_2.2.0/com.ibm.zos.v2r2.iead100/iead10060.htm

Conceptually `PSAAOLD` is an offset into the Prefixed Save Area (PSA): 
https://www.ibm.com/support/knowledgecenter/en/SSLTBW_2.2.0/com.ibm.zos.v2r2.iead300/iead300767.htm
The PSA always starts at address zero: 
https://www.ibm.com/support/knowledgecenter/zosbasics/com.ibm.zos.zconcepts/zconcepts_104.htm

So it's really a question of style, whether you consider `base+offset` used consistently to be clearer (even when base is zero) than `offset` (when base is zero) and `base+offset` everywhere else.

Reply:
the problem with using base+offset when base == NULL is that it relies in undefined behaviour and the fact that it gets simplified at compile time.

removing the psa variable and keeping the PSA_PTR value as part of the ascb assignment could accomplish both readability/consistency and reliability IMHO